cli/src/program.rs

```diff
@@ -487,19 +568,87 @@ pub fn parse_program_subcommand(
                 signers: signer_info.signers,
             }
         }
-        ("show", Some(matches)) => CliCommandInfo {
-            command: CliCommand::Program(ProgramCliCommand::Show {
-                account_pubkey: pubkey_of(matches, "account"),
-            }),
-            signers: vec![],
-        },
+        ("show", Some(matches)) => {
+            let account_pubkey = if matches.is_present("buffers") {
+                None
+            } else {
+                pubkey_of(matches, "account")
+            };
+
+            let authority_pubkey = if let Some(authority_pubkey) =
+                pubkey_of_signer(matches, "buffer_authority", wallet_manager)?
+            {
+                authority_pubkey
+            } else if let (_, Some(authority_pubkey)) =
+                signer_of(matches, "buffer_authority", wallet_manager)?
```

Comment:
I'm probably wrong but isn't this doing the same thing as `pubkey_of_signer` above?

Reply:
Yeah, I think I copypasta'd that pattern from somewhere else, will fix all those cases in a followup